### PR TITLE
feat: introduce pagination to python

### PR DESCRIPTION
**Pull request by armandobelardo**
Needs testing but this should be the bulk of it
Fixes FER-1522

This consumes the pagination IR
The generated `client.Users.List` can then be used as a `User` generator (effectively the "default"):

  ```python
  for user in client.users.list(...):
    print user
  ```

  a page-by-page generator:

  ```python
  for page in client.users.list(...).iter_pages():
    print(page.items)
  ```

  or statically calling `next_page` to perform the pagination manually:

  ```python
  pager = client.users.list(...)
  # First page
  print(pager.items)
  # Second page
  pager = pager.next_page()
  print(pager.items)
  ```

### CODE REVIEW

**Comment by dsinghvi** [comment]
The client api looks great -- can you document it [here](https://buildwithfern.com/learn/sdks/features/auto-pagination) once we finalize it?

File: generators/python/core_utilities/sdk/pagination.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+class SyncPage(BasePage, typing.Generic[T]):
+    get_next: typing.Optional[typing.Callable[[], typing.Optional[Self]]]
+
+class AsyncPage(BasePage, typing.Generic[T]):
+    get_next: typing.Optional[typing.Callable[[], typing.Awaitable[typing.Optional[Self]]]]
+
+# ----------------------------
+
+class SyncPaginator(SyncPage[T], typing.Generic[T]):
```
Comment (dsinghvi) [inline generators/python/core_utilities/sdk/pagination.py:32]: Should we align the naming here with Go like `SyncPager`, `AsyncPager` ?

File: generators/python/sdk/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,66 @@ All notable changes to this project will be documented in this file.
...
+
+  or statically calling `next_page` to perform the pagination manually:
+
+  ```python
+  pager = client.users.list(...)
+  # First page
+  print(pager.items)
+  # Second page
+  pager = pager.next_page()
+  print(pager.items)
```
Comment (dsinghvi) [inline generators/python/sdk/CHANGELOG.md:65]: @armandobelardo we'll want to gate this feature as well

File: generators/python/sdk/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-1.7.0-rc0
+2.0.1
```
Comment (dsinghvi) [inline generators/python/sdk/VERSION:1]: should we do a minor since its a feature release

File: seed/python-sdk/alias/src/seed/core/pagination.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+# This file was auto-generated by Fern from our API Definition.
+
+import asyncio
```
Comment (dsinghvi) [inline seed/python-sdk/alias/src/seed/core/pagination.py:3]: can we only include `pagination.py` if there are paginated endpoints?

File: seed/python-sdk/pagination/src/seed/users/client.py [hunk 5]
```diff
@@ -101,7 +103,21 @@ def list_with_cursor_pagination(
...
+                _has_next = _parsed_next is not None
+                _get_next = lambda: self.list_with_cursor_pagination(
+                    page=page,
+                    per_page=per_page,
+                    order=order,
+                    starting_after=_parsed_next,
+                    request_options=request_options,
+                )
+            _items = _parsed_response.data
+            return SyncPaginator(has_next=_has_next, items=_items, get_next=_get_next)
```
Comment (dsinghvi) [inline seed/python-sdk/pagination/src/seed/users/client.py:120]: how does this work if `_parsed_next` and `_has_next` aren't lambdas and aren't recomputed on every page load?

File: seed/python-sdk/pagination/src/seed/users/client.py [hunk 6]
```diff
@@ -116,7 +132,7 @@ def list_with_offset_pagination(
         order: typing.Optional[Order] = None,
         starting_after: typing.Optional[str] = None,
         request_options: typing.Optional[RequestOptions] = None,
-    ) -> ListUsersPaginationResponse:
+    ) -> SyncPaginator[User]:
```
Comment (dsinghvi) [inline seed/python-sdk/pagination/src/seed/users/client.py:135]: i don't see `snippets.json` updated for this endpoint, but would be nice to include the `for ... if`

File: generators/python/core_utilities/sdk/pagination.py (hunk 1 above)
Comment (armandobelardo) [reply generators/python/core_utilities/sdk/pagination.py:32]: Updating!

File: seed/python-sdk/pagination/src/seed/users/client.py (hunk 6 above)
Comment (armandobelardo) [reply seed/python-sdk/pagination/src/seed/users/client.py:135]: It might be easier to include in a different PR

File: seed/python-sdk/pagination/src/seed/users/client.py (hunk 5 above)
Comment (armandobelardo) [reply seed/python-sdk/pagination/src/seed/users/client.py:120]: discussed offline, but the `get_next` is the API call wrapped in a lambda, so it's passing in the function call it was just in, with updated parameters
so the full SyncPager is recomputed and returned on the `get_next` call

File: seed/python-sdk/alias/src/seed/core/pagination.py (hunk 4 above)
Comment (armandobelardo) [reply seed/python-sdk/alias/src/seed/core/pagination.py:3]: Will add!

File: generators/python/sdk/CHANGELOG.md (hunk 2 above)
Comment (armandobelardo) [reply generators/python/sdk/CHANGELOG.md:65]: Will add a flag, I think it might be easier to just omit from the IR, but will think about it more

Comment (armandobelardo) [reply generators/python/sdk/CHANGELOG.md:65]: PR here: https://github.com/fern-api/venus/pull/150

File: test-definitions/fern/apis/pagination/ir.json [hunk 7]
```diff
@@ -0,0 +1,6992 @@
+{
```
Comment (dsinghvi) [inline test-definitions/fern/apis/pagination/ir.json:1]: is this file intentional?

**Review by dsinghvi — APPROVED** [review]
looks great -- can we run `yarn seed:build and yarn seed write-inputs` for all of the generators. 

And can we make this a `2.1.0` bump

File: test-definitions/fern/apis/pagination/ir.json (hunk 7 above)
Comment (armandobelardo) [reply test-definitions/fern/apis/pagination/ir.json:1]: ah no it's not! removing
